src/main/java/io/appium/java_client/pagefactory/bys/builder/ByAll.java

```diff
@@ -0,0 +1,52 @@
+package io.appium.java_client.pagefactory.bys.builder;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.SearchContext;
+import org.openqa.selenium.WebElement;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+
+
+public class ByAll extends org.openqa.selenium.support.pagefactory.ByAll {
+
+    private final List<By> bys;
+
+    private Function<SearchContext, Optional<WebElement>> getSearchingFunction(By by) {
+        return input -> {
+            try {
+                return Optional.of(input.findElement(by));
+            } catch (NoSuchElementException e) {
+                return Optional.empty();
+            }
+        };
+    }
+
+    /**
+     * @param bys is a set of {@link org.openqa.selenium.By} which forms the all possible searching.
+     */
+    public ByAll(By[] bys) {
+        super(bys);
+        checkNotNull(bys);
+
+        this.bys = Arrays.asList(bys);
+        if (this.bys.isEmpty()) {
```

Comment:
you can also use http://google.github.io/guava/releases/snapshot/api/docs/com/google/common/base/Preconditions.html#checkArgument(boolean) since _checkNotNull_ is already there